Production/python/TauL2Prod.py

```diff
@@ -0,0 +1,185 @@
+
+options = VarParsing('analysis')
+options.register('sampleType', '', VarParsing.multiplicity.singleton, VarParsing.varType.string,
+                 "Indicates the sample type: Summer16MC, Run2016, ...")
+options.register('fileList', '', VarParsing.multiplicity.singleton, VarParsing.varType.string,
+                 "List of root files to process.")
+options.register('fileNamePrefix', '', VarParsing.multiplicity.singleton, VarParsing.varType.string,
+                 "Prefix to add to input file names.")
+options.register('tupleOutput', 'eventTuple.root', VarParsing.multiplicity.singleton, VarParsing.varType.string,
+                 "Event tuple file.")
+options.register('lumiFile', '', VarParsing.multiplicity.singleton, VarParsing.varType.string,
+                 "JSON file with lumi mask.")
+options.register('eventList', '', VarParsing.multiplicity.singleton, VarParsing.varType.string,
+                 "List of events to process.")
+options.register('dumpPython', False, VarParsing.multiplicity.singleton, VarParsing.varType.bool,
+                 "Dump full config into stdout.")
+options.parseArguments()
+
+isData = False
+if 'Data' in options.sampleType:
+    isData = True
+print isData
+processName = 'reHLT'
+process = cms.Process(processName, Run3)
+
+# import of standard configurations
+process.load('Configuration.StandardSequences.Services_cff')
+process.load('SimGeneral.HepPDTESSource.pythiapdt_cfi')
+process.load('FWCore.MessageService.MessageLogger_cfi')
+process.load('Configuration.EventContent.EventContent_cff')
+process.load('Configuration.StandardSequences.GeometryRecoDB_cff')
+process.load('Configuration.StandardSequences.MagneticField_cff')
+process.load('HLTrigger.Configuration.HLT_User_cff')
+process.load('Configuration.StandardSequences.EndOfProcess_cff')
+process.load('Configuration.StandardSequences.FrontierConditions_GlobalTag_cff')
+
+if not isData:
+    process.load('SimGeneral.MixingModule.mixNoPU_cfi')
+
+from TauMLTools.Production.readFileList import *
+if len(options.fileList) > 0:
+    readFileList(process.source.fileNames, options.fileList, options.fileNamePrefix)
+elif len(options.inputFiles) > 0:
+    addFilesToList(process.source.fileNames, options.inputFiles, options.fileNamePrefix)
+
+if len(options.lumiFile) > 0:
+    import FWCore.PythonUtilities.LumiList as LumiList
+    process.source.lumisToProcess = LumiList.LumiList(filename = options.lumiFile).getVLuminosityBlockRange()
+
+if options.eventList != '':
+    process.source.eventsToProcess = cms.untracked.VEventRange(re.split(',', options.eventList))
+
+
+process.maxEvents = cms.untracked.PSet(
+    input = cms.untracked.int32(options.maxEvents),
+    output = cms.optional.untracked.allowed(cms.int32,cms.PSet)
+)
+
+# Input source
+if isData:
+    process.source = cms.Source("PoolSource",
+        fileNames = cms.untracked.vstring('/store/data/Run2018D/EphemeralHLTPhysics1/RAW/v1/000/325/113/00000/EA32A985-9BAF-D044-96FF-D59033C22A09.root'),
```

Comment:
- don't put hardcoded filenames here and below. 
- `process.source` should be defined before processing `options.fileList`, `options.inputFiles`, `options.lumiFile` and `options.eventList`
- `if isData` is not necessary for `process.source` definition